lib/dec/tail.c

```diff
@@ -30,10 +30,10 @@
 		}
 		if ((q = base64_table_dec[*c++]) >= 254) {
 			st.eof = 1;
-			// When q == 254, the input char is '='. Return 1 and EOF.
-			// Technically, should check if next byte is also '=', but never mind.
+			// When q == 254, the input char is '='.
+			// Check if next byte is also '=', Return 1 and EOF.
 			// When q == 255, the input char is invalid. Return 0 and EOF.
-			ret = (q == 254) ? 1 : 0;
+			ret = ((q == 254) && (base64_table_dec[*c++] == 254)) ? 1 : 0;
```

Comment:
This is incorrect; the loop may only look at one byte at a time. It cannot assume that the next byte exists.

One way of checking if there is a byte available is whether `srclen` is larger than 0, but even then you need to handle the case where there is no next byte (yet).